### PR TITLE
support read xvector egs from python

### PR DESCRIPTION
**Pull request by fanlu**

### CODE REVIEW

File: src/pybind/kaldi/table.py [hunk 1]
```diff
@@ -190,6 +193,11 @@ class SequentialNnetChainExampleReader(_SequentialReaderBase,
     '''Sequential table reader for nnet chain examples.'''
     pass
 
+class SequentialNnetExampleReader(_SequentialReaderBase,
```
Comment (csukuangfj) [inline src/pybind/kaldi/table.py]: Line 575 in this file can be removed.

File: src/pybind/kaldi/table.py [hunk 2]
```diff
@@ -342,6 +350,10 @@ class RandomAccessNnetChainExampleReader(_RandomAccessReaderBase,
     '''Random access table reader for nnet chain examples.'''
     pass
 
+class RandomAccessNnetExampleReader(_RandomAccessReaderBase,
```
Comment (csukuangfj) [inline src/pybind/kaldi/table.py]: Line 661 in this file can also be removed.

File: src/pybind/nnet3/nnet_example_pybind.cc [hunk 3]
```diff
@@ -40,4 +41,19 @@ void pybind_nnet_example(py::module& m) {
             "SparseMatrix would be the natural format for posteriors).");
     // TODO(fangjun): other constructors, fields and methods can be wrapped when
   }
+  {
+    using PyClass = NnetExample;
+    py::class_<PyClass>(m, "NnetExample")
```
Comment (csukuangfj) [inline src/pybind/nnet3/nnet_example_pybind.cc]: You can copy and paste the documentation from c++ to pybind.

For example,

```cpp
py::class_<PyClass>(m, "NnetExample", "NnetExample is the input data....")
```

Later, we will generate static HTML documentation from these comments.

**Comment by csukuangfj** [comment]
I think only your last commit is relevant to this pullrequest. You can **rebase** your local branch
to the latest kaldi pybind11 branch before committing.

File: src/pybind/nnet3/nnet_example_pybind.cc [hunk 4]
```diff
@@ -40,4 +41,19 @@ void pybind_nnet_example(py::module& m) {
             "SparseMatrix would be the natural format for posteriors).");
     // TODO(fangjun): other constructors, fields and methods can be wrapped when
   }
+  {
```
Comment (csukuangfj) [inline src/pybind/nnet3/nnet_example_pybind.cc]: I think you can also add author info to the beginning of the file.

File: src/pybind/kaldi/table.py (hunk 1 above)
Comment (fanlu) [reply src/pybind/kaldi/table.py]: ok

File: src/pybind/kaldi/table.py (hunk 2 above)
Comment (fanlu) [reply src/pybind/kaldi/table.py]: ok

File: src/pybind/nnet3/nnet_example_pybind.cc (hunk 3 above)
Comment (fanlu) [reply src/pybind/nnet3/nnet_example_pybind.cc]: I'll add it

**Comment by fanlu** [comment]
Done, please hive a review @csukuangfj

**Comment by csukuangfj** [comment]
thanks, I'll merge when the CI passes.

**Comment by csukuangfj** [comment]
merging.

**Pull request merged**